libs/design/switch/src/switch/switch.component.ts

```diff
@@ -0,0 +1,89 @@
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  Output,
+} from '@angular/core';
+
+import { DAFF_LOADING_ICON_COMPONENTS } from '@daffodil/design/loading-icon';
+
+import {
+  DaffLabelPosition,
+  DaffLabelPositionEnum,
+} from './label-position';
+import { DaffSwitchErrorMessage } from './switch-errors';
+
+let switchUniqueId = 0;
+
+@Component({
+  selector: 'daff-switch',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './switch.component.html',
+  styleUrls: ['./switch.component.scss'],
+  standalone: true,
+  imports: [
+    DAFF_LOADING_ICON_COMPONENTS,
+  ],
+})
+export class DaffSwitchComponent {
+  @Input() checked = false;
+  @Input() loading = false;
+  @Input() labelPosition: DaffLabelPosition = DaffLabelPositionEnum.LEFT;
+
+  @HostBinding('attr.tabindex') tabindex = 0;
+
+  @HostListener('keydown', ['$event'])
+  handleKeydown(event: KeyboardEvent) {
+    if (event.code === 'Space') {
+      event.preventDefault();
+      this.onToggle();
+    }
+  }
+
+  /**
+   * @docs-private
+   */
+  @HostBinding('attr.aria-label') private externalAriaLabel = null;
+
+  /**
+   * aria-label for the switch.
+   */
+  @Input('aria-label') ariaLabel = '';
+
+  @Input() id: string = 'daff-switch-' + switchUniqueId++;
+
+  @Output() toggle = new EventEmitter<boolean>();
+
+  onToggle() {
+    if (!this.disabled) {
+      try {
+        this.checked = !this.checked;
+        this.toggle.emit(this.checked);
+      } catch (error) {
+        throw new Error(DaffSwitchErrorMessage);
```

Comment:
I don't think we need to try and catch the error here? Should be up to the implementation

i.e.
```
<daff-switch [checked]="checked" [formControl]="defaultSwitchExample">
  Default
  <daff-error-message *ngIf="defaultSwitchExample.invalid">Custom error message</golf-error-message>
</daff-switch>
```

Reply:
I made updates here [0e182c5](https://github.com/graycoreio/daffodil/pull/3143/commits/0e182c51aab72c81a575367fdeea56b5ab534b50) to how the error message should be used.

Also, switches that have errored should still be toggle-able.